Detectors/ZDC/simulation/include/ZDCSimulation/Digits2Raw.h

```diff
@@ -0,0 +1,103 @@
+#include <bitset>
+#include <Rtypes.h>
+#include "Headers/RAWDataHeader.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "DetectorsRaw/RawFileWriter.h"
+#include "DetectorsRaw/HBFUtils.h"
+#include "DataFormatsZDC/RawEventData.h"
+#include "ZDCBase/ModuleConfig.h"
+#include "ZDCSimulation/SimCondition.h"
+#include "DataFormatsZDC/BCData.h"
+#include "DataFormatsZDC/ChannelData.h"
+
+namespace o2
+{
+namespace zdc
+{
+class Digits2Raw
+{
+ public:
+  Digits2Raw() = default;
+  void emptyBunches(std::bitset<3564>& bunchPattern); /// Prepare list of clean empty bunches for baseline evaluation
+  void processDigits(const std::string& outDir, const std::string& fileDigitsName);
+  void setModuleConfig(const ModuleConfig* moduleConfig) { mModuleConfig = moduleConfig; };
+  const ModuleConfig* getModuleConfig() { return mModuleConfig; };
+  void setSimCondition(const SimCondition* SimCondition) { mSimCondition = SimCondition; };
+  const SimCondition* getSimCondition() { return mSimCondition; };
+
+  o2::raw::RawFileWriter& getWriter() { return mWriter; }
+
+  void setFilePerLink(bool v) { mOutputPerLink = v; }
+  bool getFilePerLink() const { return mOutputPerLink; }
+
+  void setVerbosity(int v)
+  {
+    mVerbosity = v;
+    mWriter.setVerbosity(v);
+  }
+  int getVerbosity() const { return mVerbosity; }
+
+  //  void setContinuous(bool v = true) { mIsContinuous = v; }
+  bool isContinuous() const { return mIsContinuous; }
+  static void print_gbt_word(const UInt_t* word, const ModuleConfig* moduleConfig = 0);
+
+ private:
+  void setTriggerMask();
+  void updatePedestalReference(int bc);
+  void resetSums(uint32_t orbit);
+  void resetOutputStructure(UShort_t bc, UInt_t orbit, bool is_dummy);       /// Reset output structure not incrementing scalers for dummy bunches
+  void assignTriggerBits(int ibc, UShort_t bc, UInt_t orbit, bool is_dummy); /// Assign trigger bits
+  void insertLastBunch(int ibc, uint32_t orbit);                             /// Insert an empty bunch at last position in orbit
+  void convertDigits(int ibc);                                               /// Convert digits into raw data
+  void writeDigits();                                                        /// Writes raw data to file
+  std::vector<o2::zdc::BCData> mzdcBCData, *mzdcBCDataPtr = &mzdcBCData;
+  std::vector<o2::zdc::ChannelData> mzdcChData, *mzdcChDataPtr = &mzdcChData;
+  int mNbc = 0;
+  BCData mBCD;
+  EventData mZDC;                                                       /// Output structure
+  bool mIsContinuous = true;                                            /// Continuous (self-triggered) or externally-triggered readout
+  bool mOutputPerLink = false;                                          /// Split output
+  const ModuleConfig* mModuleConfig = 0;                                /// Trigger/readout configuration object
+  const SimCondition* mSimCondition = 0;                                /// Pedestal/noise configuration object
```

Comment:
the new standard requires ``nullptr`` instead of 0. Due to this you get errors in https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/4658/bff9a440648e881ac50c8acce1a06c40eba16893/build_O2_fullCI/pretty.html

Please fix all of them.